docs/developers/components/reference.md

```diff
@@ -97,21 +97,491 @@ If a `config.yaml` is defined, it will **not** be merged with the default config
 
 A Harper Extension is a extensible component that is intended to be used by other components. The built-in components [graphqlSchema](./built-in.md#graphqlschema) and [jsResource](./built-in.md#jsresource) are both examples of extensions.
 
-There are two key types of Harper Extensions: **Resource Extension** and **Protocol Extensions**. The key difference is a **Protocol Extensions** can return a **Resource Extension**.
+> As of Harper v4.6, a new Extension API has been introduced as a major overhaul of the previous API. The new API is designed to be more flexible, extensible, performant, and easier to use. It is recommended that all new extensions use the new API, and that existing extensions are migrated to the new API as soon as possible. The documentation for the legacy API is still available below in the [Legacy Extensions](#legacy-extensions) section.
 
-Functionally, what makes an extension a component is the contents of `config.yaml`. Unlike the Application Template referenced earlier, which specified multiple components within the `config.yaml`, an extension will specify an `extensionModule` option.
-
-- **extensionModule** - `string` - _required_ - A path to the extension module source code. The path must resolve from the root of the extension module directory.
+Functionally, what makes an extension a component is the contents of `config.yaml`. Unlike the Application Template referenced earlier, which specified multiple components within the `config.yaml`, an extension must specify an `extensionModule` option. This must be a path to the extension module source code. The path must resolve from the root of the extension module directory.
 
 For example, the [Harper Next.js Extension](https://github.com/HarperDB/nextjs) `config.yaml` specifies `extensionModule: ./extension.js`.
 
 If the extension is being written in something other than JavaScript (such as TypeScript), ensure that the path resolves to the built version, (i.e. `extensionModule: ./dist/index.js`)
 
 It is also recommended that all extensions have a `package.json` that specifies JavaScript package metadata such as name, version, type, etc. Since extensions are just JavaScript packages, they can do anything a JavaScript package can normally do. It can be written in TypeScript, and compiled to JavaScript. It can export an executable (using the [bin](https://docs.npmjs.com/cli/configuring-npm/package-json#bin) property). It can be published to npm. The possibilities are endless!
 
-Furthermore, what defines an extension separately from a component is that it leverages any of the [Resource Extension](#resource-extension-api) or [Protocol Extension](#protocol-extension-api) APIs. The key is in the name, **extensions are extensible**.
+Furthermore, what defines an extension separately from a component is that it exports a `handleComponent()` method.
```

Comment:
I'm sure this has been discussed previously and there are good reasons for it, but it is a little surprising that the thing that defines an extension separately from a component is exporting a method named `handleComponent` (vs. `handleExtension`). Would it make sense to add a short explanation here?

Reply:
I will continue to iterate on clarifying the definitions of Components, Applications, and Extensions in a follow up PR